### PR TITLE
RAFD-2563: Changes for dataprep

### PR DESCRIPTION
**Pull request by prashant-gour-guavus**

### CODE REVIEW

File: wrangler-service/src/main/java/co/cask/wrangler/PropertyIds.java [hunk 1]
```diff
@@ -64,4 +64,17 @@
   public static final String VALUE_DESERIALIZER = "value.deserializer";
 
   public static final String PLUGIN_SPECIFICATION = "plugin.specification";
+  
+  // Key tab pricipal to be used
+  public static final String PRINCIPAL = "principal";
+  
+  // Location of tab file
+  public static final String KEYTAB_LOCATION = "keytabLocation";
+
```
Comment (zuxqoj) [inline wrangler-service/src/main/java/co/cask/wrangler/PropertyIds.java:73]: Fetch keytab details from namespace.

File: wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java [hunk 2]
```diff
@@ -280,16 +286,24 @@ public void specification(HttpServiceRequest request, final HttpServiceResponder
...
 
       Map<String, String> properties = new HashMap<>();
-      properties.put("topic", topic);
+      properties.put(PropertyIds.TOPIC, topic);
       properties.put("referenceName", topic);
       properties.put("brokers", conn.getProp(PropertyIds.BROKER));
       properties.put("kafkaBrokers", conn.getProp(PropertyIds.BROKER));
       properties.put("keyField", conn.getProp(PropertyIds.KEY_DESERIALIZER));
-      properties.put("format", "text");
+      properties.put(PropertyIds.FORMAT, "text");
```
Comment (zuxqoj) [inline wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java]: do we want to support `text` type only ?

File: wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaConfiguration.java [hunk 3]
```diff
@@ -16,74 +16,140 @@
...
+    }
 
+    if (kafkaProducerProperties.containsKey(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG)) {
+        valueDeserializer = kafkaProducerProperties.get(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG);
+    }
+    
+    if (kafkaProducerProperties.containsKey(ConsumerConfig.EXCLUDE_INTERNAL_TOPICS_CONFIG)) {
+        excludeInternalTopic = kafkaProducerProperties.get(ConsumerConfig.EXCLUDE_INTERNAL_TOPICS_CONFIG);
+    }
+
```
Comment (zuxqoj) [inline wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaConfiguration.java]: preferably use getOrElse

File: wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaConfiguration.java [hunk 4]
```diff
@@ -16,74 +16,140 @@
...
+				"        useKeyTab=true \n" +
+				"        storeKey=true  \n" +
+				"        useTicketCache=false  \n" +
+				"        renewTicket=true  \n" +
+				"        keyTab=\"%s\" \n" +
+				"        principal=\"%s\";",
+				keytabLocation, principal));
+	}
+	
+	props.forEach((k,v)->LOG.info("Item : {}, Value :  {}", k, v));
```
Comment (zuxqoj) [inline wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaConfiguration.java]: change log level to debug

File: wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java [hunk 5]
```diff
@@ -274,22 +284,40 @@ public void read(HttpServiceRequest request, HttpServiceResponder responder,
...
+      }
 
-      kafka.add("properties", gson.toJsonTree(properties));
-      kafka.addProperty("name", "Kafka");
+      if (conn.getAllProps().containsKey(PropertyIds.KAFAK_PRODUCER_PROPERTIES)) {
+    	  Type type = new TypeToken<HashMap<String, String>>(){}.getType();
+    	  Map<String, String> kafkaProducerProperties = gson.fromJson(conn.getAllProps().get(PropertyIds.KAFAK_PRODUCER_PROPERTIES), type);
+    	  properties.put(PropertyIds.KAFKA_PROPERTIES, Joiner.on(",").withKeyValueSeparator(":").join(kafkaProducerProperties));
+    	  
+    	  if (kafkaProducerProperties.containsKey(PropertyIds.FORMAT) && Arrays.asList("text", "tsv", "binary", "avro", "csv").contains(kafkaProducerProperties.get(PropertyIds.FORMAT))) {
```
Comment (zuxqoj) [inline wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java]: add toLower 
kafkaProducerProperties.get(PropertyIds.FORMAT).toLower to make it case-insensitive if this value is not coming from any drop-down

Comment (zuxqoj) [inline wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java]: we should support avro schema registry , you can create a separate ticket for that

**Review by zuxqoj — CHANGES_REQUESTED** [review]

**Comment by zuxqoj** [comment]
don't we need to add/update UTs for this change ??

File: wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java (hunk 5 above)
Comment (prashant-gour-guavus) [reply wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java]: https://guavus-jira.atlassian.net/browse/RAFD-3029

Comment (prashant-gour-guavus) [reply wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java]: Adding lowercase

**Review by zuxqoj — APPROVED** [review]

**Pull request merged**